### PR TITLE
feat(curriculum): create javascript strings quiz

### LINKED ISSUE #56197: [HACKTOBERFEST 2024] - Strings Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Strings quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-javascript-strings`.

**All quiz questions need to relate to one of these topics:**

- bracket notation (and extracting character form string)
- template literals and interpolation
- escape sequences and new lines
- String constructor
- substrings
- `prompt()` method
- string object vs primitive
- ASCII and charCodeAt() and fromCharCode()
- toString()
- includes()
- slice()
- toLowerCase() toUpperCase()
- replace()
- repeat()
- trim()

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by bytexenon**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #56197

### CODE REVIEW

**Review by gikf — COMMENTED** [review]
Also note for future - it is recommended to make changes on a separate branch, rather than the `main` branch. See more about it in point 3 at https://contribute.freecodecamp.org/how-to-contribute-to-the-codebase/#contributing-to-the-codebase.

**Review by gikf — APPROVED** [review]

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 1]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is the purpose of the `prompt()` function?
 
 #### --distractors--
 
-Placeholder distractor 1
+Displays a message in the console
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:156]:
```suggestion
It displays a message in the console.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 2]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+Displays an alert box with a message
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:160]:
```suggestion
It displays an alert box with a message.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 3]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+Displays a confirmation box with a message
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:164]:
```suggestion
It displays a confirmation box with a message.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 4]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+Displays a dialog box that prompts the user for input
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:168]:
```suggestion
It displays a dialog box that prompts the user for input.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 5]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --text--
 
-Placeholder question
+What is the difference between a string object and a string primitive?
 
 #### --distractors--
 
-Placeholder distractor 1
+String objects have methods, string primitives do not
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:178]:
```suggestion
String objects have methods, string primitives do not.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 6]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 2
+String primitives are created using the `String` constructor, string objects are created using string literals
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:182]:
```suggestion
String primitives are created using the `String` constructor, string objects are created using string literals.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 7]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+There is no difference between a string object and a string primitive
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:186]:
```suggestion
There is no difference between a string object and a string primitive.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 8]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+String objects are created using the `String` constructor, string primitives are created using string literals
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:190]:
```suggestion
String objects are created using the `String` constructor, string primitives are created using string literals.
```

**Review by majestic-owl448 — COMMENTED** [review]
Sentences need to finish with a period

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 9]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+`'Hello, ' + user + '!'`
 
 ---
 
-Placeholder distractor 2
+`'Hello, ${user}!'`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:50]:
```suggestion
`'Hello, $user!'`
```

As is, this answer *is* valid string interpolation.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md [hunk 10]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+`'JavaScript'.extract(0, 3)`
 
 #### --answer--
 
-Placeholder answer
+`'JavaScript'.substr(0, 3)`
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:146]: `.substr()` is deprecated. We should probably use `.substring()`.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This looks good! A couple of small nits.

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md (hunk 10 above)
Comment (ByteXenon) [reply curriculum/challenges/english/25-front-end-development/quiz-javascript-strings/66edc31c44f1b9c1d5c5ebca.md:146]: Thank you for pointing that out. I've completely forgot about the deprecation of `.substr()`. I will update the code to use `.substring()` instead.

**Comment by ByteXenon** [comment]
All issues have been fixed, waiting for the feedback!

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**